src/renderer/src/App.tsx

```diff
@@ -1,12 +1,52 @@
 import Versions from './components/Versions'
+import { IntlConfig, IntlProvider } from 'react-intl'
+import { createContext, Dispatch, SetStateAction, useEffect, useMemo, useState } from 'react'
 
-function App(): JSX.Element {
-  return (
-    <div className="container">
-      <Versions></Versions>
-      {/* Render other stuff here! */}
-    </div>
-  )
+import * as es from '../translations/es.json'
+import * as en from '../translations/en.json'
+
+const DEFAULT_LOCALE = 'es'
+
+type Translations = {
+  es: IntlConfig['messages']
+  en: IntlConfig['messages']
+}
+
+const translations: Translations = { es, en }
+type AvailableLocales = keyof typeof translations
+
+const isTranslation = (langugage?: string): langugage is AvailableLocales => {
+  if (!langugage) return false
+  return Object.keys(translations).includes(langugage)
+}
+
+type IntlSetContextType = Readonly<[string, Dispatch<SetStateAction<AvailableLocales>>]>
+
+export const IntlSwitchConext = createContext<IntlSetContextType>([DEFAULT_LOCALE, (): void => {}])
+
+function App(): JSX.Element | null {
+  const [lang, setLang] = useState<AvailableLocales>(DEFAULT_LOCALE)
+  useEffect(() => {
+    setupApp()
+  }, [])
+
+  const setupApp = async (): Promise<void> => {
+    const locale = await window.api.getLocale()
+    setLang(isTranslation(locale) ? locale : DEFAULT_LOCALE)
+  }
```

Comment:
This is going to cause a quick flash where it is set to default locale, detects a different language, and then sets the different language.

We should add this logic into the loading state (either the splash screen or add a loader to this page)

Secondly, `useEffect` best practices says that we should add `setupApp` as a dependecy, which will require you to `useCallback` setupApp. So it would probably be easier to just define it inside the `useEffect`

Eg:
```ts
 useEffect(() => {
    setupApp()

   const setupApp = async (): Promise<void> => {
        const locale = await window.api.getLocale()
        setLang(isTranslation(locale) ? locale : DEFAULT_LOCALE)
  }
  }, [])